projects/graph/src/graph.py

```diff
@@ -80,6 +80,35 @@ def bft(self, starting_vertex_id):
                 for neighbor in self.vertices[v]:
                     q.enqueue(neighbor)
 
+    def bfs(self, starting_vertex_id, query):
+        # Create an empty queue
+        q = Queue()
+        # Create an empty set of visited vertices
+        visited = set()
+
+        # Put the starting vertex in our Queue
+        q.enqueue(starting_vertex_id)
+        # While the queue is not empty....
+
+        def bfs_helper(self, starting_id):
```

Comment:
Interesting use of helper function here.  When you have a moment, try to understand also the while loop implementation in the model solution so you have more arsenals in your tool kit in the future.

Reply:
yes.  I changed it so there is no helper function :)